api-registration/server/index.js

```diff
@@ -0,0 +1,53 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+  insert into users (username, "hashedPassword")
+  values ($1, $2)
+  returning username, "userId", "createdAt"
+  `;
+      const params = [username, hashedPassword];
+
+      db.query(sql, params)
+        .then(result => {
+          const newRow = result.rows;
```

Comment:
`newRow` will be an array, so you are sending an array back to the client and the returned value should be an object. You can use destucturing or indicate the 0 index

Example:
```JAVASCRIPT
const [ newRow ] = result.rows;

// --OR --

const newRow = result.rows[0];
```